fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1700,6 +1707,7 @@ Future<Void> ReadYourWritesTransaction::watch(const Key& key) {
 	if (key.size() > (key.startsWith(systemKeys.begin) ? CLIENT_KNOBS->SYSTEM_KEY_SIZE_LIMIT : CLIENT_KNOBS->KEY_SIZE_LIMIT))
 		return key_too_large();
 
+	approximateSize += key.expectedSize() + sizeof(Watch);
```

Comment:
Looking into this, it appears the cost of a watch in the commit request is one read conflict range. That means that you shouldn't add anything to size unless we are bypassing RYW, in which case you should add the cost of the conflict range.